File: Js/src/indice.js
//calculadora imc

import { encabezado, line, readLineNumber } from "./utilidades.js";

const main = () => {

  while (true) {
    encabezado('Calculadora IMC', 32)
    let peso = readLineNumber('Agrega tu peso (kg)')
    let altura = readLineNumber('Agrega tu altura (mts)')

    let imc = resultado(peso, altura)

    if (isNaN(imc) || imc == null) {
      console.error('Error intenta de nuevo')
    } else {
      console.log('tu IMC es: ' + imc.toFixed(2))
      line(32)
      console.log(categorizarIMC(imc))
      line(32)
      break;
    }

  }
}

const resultado = (peso = 0, altura = 0, result = 0) => {
  return result = (peso / (altura * altura))
}

const categorizarIMC = (imc) => {
  if (imc < 18.5) return 'Bajo peso';
  if (imc < 24.9) return 'Peso normal';
  if (imc < 29.9) return 'Sobrepeso';
  return 'Obesidad';
}

main()

File: python/indice.py
'''
Calculadora de IMC
'''

from utilidades import linea, stringToNumber, encabezado
#Aqui importamos lo q se utilizara
  

def main () -> None:
  #definir variables
  altura:float
  peso:float
  imc:float = 0 
  #Para que no contenga basura se inicializa en cero y liberamos ram
  encabezado('Calculadora IMC', 40)
  peso = stringToNumber('Agrega tu peso (KG) ', float)
  altura = stringToNumber('Agrega tu altura (M) ', float)
  
  imc = calculo_IMC(peso, altura)
  
  linea(40)
  print (f'Tu indice de masa corporal es {imc:.2f} ({(logica_IMC(imc).upper())})')
  
def calculo_IMC(peso: float, altura: float) -> float:
 return peso / pow(altura, 2)
  
def logica_IMC(imc: float) -> str:
  if imc < 18.5:
    return 'tienes que comer un poco mas'
  elif imc >= 18.5 and imc < 24.99:
    return 'Estas bien'
  elif imc > 25 and imc < 29.99:
    return 'estas sobre peso'
  else:
    return 'obesidad'

main()  #siempre se usa para inicializar el programa